pkg/network/telemetry/telemetry_manager.go

```diff
@@ -0,0 +1,56 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+//go:build linux_bpf
+// +build linux_bpf
+
+package telemetry
+
+import (
+	"io"
+
+	"github.com/DataDog/datadog-agent/pkg/network/ebpf/probes"
+	manager "github.com/DataDog/ebpf-manager"
+	"github.com/cilium/ebpf"
+)
+
+type ManagerWithTelemetry struct {
+	*manager.Manager
+	bpfTelemetry *EBPFTelemetry
+}
+
+func NewManagerWithTelemetry(mgr *manager.Manager, bt *EBPFTelemetry) *ManagerWithTelemetry {
+	return &ManagerWithTelemetry{
+		Manager:      mgr,
+		bpfTelemetry: bt,
+	}
+}
+
+func (m *ManagerWithTelemetry) InitManagerWithTelemetry(bytecode io.ReaderAt, opts manager.Options) error {
```

Comment:
Is there a reason for not naming this `InitWithOptions`?
By changing this to `InitWithOptions` I see 2 main benefits:
1. `ManagerWithTelemetry` becomes a drop-in replacement for `ebpf.Manager`;
2. It enforces the correct use of the library. Otherwise someone can inadvertently create a `ManagerWithTelemetry` but keep calling `InitWithOptions` which would result in a weird state

Reply:
The reason I was forced to change the name was because, keeping `InitWithOptions` as the function name causes an infinite recursion. The reason being that `ebpf.Manager` is an embedded field in `ManagerWithTelemetry`. 

This can be avoid by making `ebpf.Manager` non-embedded, but then we either have to implement wrappers for all the functions exposed by `ebpf.Manager`, or `ManagerWithTelemetry` is no longer a drop in replacement.

Is there another way to accomplish the same effect?

Reply:
Hmm, this shouldn't be an issue :thinking: 
See this example: https://go.dev/play/p/XvAJICxpXlC

Reply:
Ahhh. My bad. I had not fixed the call to `InitWithOptions` inside the wrapper function. The recursion makes more sense now!